llms/mlx_lm/server.py

```diff
@@ -46,7 +46,7 @@ def stopping_criteria(
             and how many tokens should be trimmed from the end if it has (`trim_length`).
     """
     if tokens and tokens[-1] == eos_token_id:
-        return StopCondition(stop_met=True, trim_length=0)
+        return StopCondition(stop_met=True, trim_length=1)
```

Comment:
Avoid sending the EOS token back to the client. Otherwise, we will have to manually enable skipping special tokens in decoding (https://huggingface.co/docs/transformers/main_classes/tokenizer#transformers.PreTrainedTokenizer.decode.skip_special_tokens)